### PR TITLE
Add SidebarBottom to toggle sidebar filtering based on story status

### PR DESCRIPTION
**Pull request by ghengeveld**
Fixes AP-3301, AP-3302, AP-3303

Also removes the "pending" story status, because it shows as an ugly "clock" icon and we don't _really_ need to show the "in progress" state of a test in the sidebar.

Note: this depends on a canary version of Storybook and I had to downgrade to React 16 for Storybook compatibility.

Screenshot 2023-08-23 at 16 09 59

<details>
  <summary>📦 Published PR as canary version: <code>0.0.40--canary.45.460fa88.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromaui/addon-visual-tests@0.0.40--canary.45.460fa88.0
  # or 
  yarn add @chromaui/addon-visual-tests@0.0.40--canary.45.460fa88.0
  ```
</details>

### CODE REVIEW

**Review by weeksling — APPROVED** [review]
lgtm. Question regarding the new fixed version of storybook, and whether this should wait until that's stable to be merged

**Comment by weeksling** [comment]
@ghengeveld I approved from the look of the code, however testing this I couldn't actually see the filter menu in storybook. Is there anything extra you need to do to enable or, or should switching to this branch, running `yarn` and running `yarn start` show the new slot?

File: src/utils/testsToStatusUpdate.ts [hunk 1]
```diff
@@ -7,7 +7,6 @@ export type StatusUpdate = Parameters<API["experimental_updateStatus"]>[1];
 type StoryStatus = StatusUpdate[any]["status"];
 
 export const statusMap: Partial<Record<TestStatus, StoryStatus>> = {
-  [TestStatus.InProgress]: "pending",
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply src/utils/testsToStatusUpdate.ts:10]: Agreed, I thought that we were going to show a grey pulsing dot on in progress tests.

File: package.json [hunk 2]
```diff
@@ -104,10 +104,10 @@
     "prettier": "^2.8.8",
     "prompts": "^2.4.2",
     "prop-types": "^15.8.1",
-    "react": "^18.0.0",
-    "react-dom": "^18.0.0",
+    "react": "^16.14.0",
+    "react-dom": "^16.14.0",
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply package.json:108]: Storybook is planning up upgrading to `react@18` as part of the 8.0 release at the end of the year.

File: src/SidebarBottom.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    !active &&
+    css({
+      "&:hover": {
+        color: theme.color.defaultText,
+      },
+    })
+);
+
+const filterNone: API_FilterFunction = () => true;
+const filterWarn: API_FilterFunction = ({ status }) => status?.[ADDON_ID]?.status === "warn";
```
Comment (tmeasday) [inline src/SidebarBottom.tsx:34]: What about errors?

**Review by tmeasday — APPROVED** [review]
Code LGTM but we should work out what we are doing with errors. I'm OK with creating a ticket for that and merging as is though.

File: src/components/SuffixInput.tsx [hunk 4]
```diff
@@ -49,14 +49,24 @@ const SuffixOverlay = ({
...
   return (
     <InputWrapper>
-      <Input id={id} hideLabel label="" value={value} placeholder={placeholder} {...props} />
+      <Input
+        id={id}
+        hideLabel
+        label=""
+        value={value}
+        placeholder={placeholder}
+        crossOrigin={undefined}
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply src/components/SuffixInput.tsx:66]: Nope, it's a React 16 thing.

File: src/utils/testsToStatusUpdate.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghengeveld) [reply src/utils/testsToStatusUpdate.ts:10]: I would love to but not if it looks that awful.
@ndelangen can you fix the "pending" icon to "look pretty"?

File: src/SidebarBottom.tsx (hunk 3 above)
Comment (ghengeveld) [reply src/SidebarBottom.tsx:34]: That doesn't seem to be a thing.

Screenshot 2023-08-24 at 14 57 33

File: src/utils/testsToStatusUpdate.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (weeksling) [reply src/utils/testsToStatusUpdate.ts:10]: I think @MichaelArestad is going to be making some improvements to ui including those icons

[the remark this answers is not in the document]
Comment (MichaelArestad) [reply src/utils/testsToStatusUpdate.ts:10]: I am planning on working with @ndelangen on that.

File: package.json [hunk 5]
```diff
@@ -69,23 +69,23 @@
     "@graphql-codegen/cli": "^4.0.1",
     "@graphql-codegen/client-preset": "^4.0.1",
     "@graphql-typed-document-node/core": "^3.2.0",
-    "@storybook/addon-actions": "^7.3.0",
-    "@storybook/addon-essentials": "^7.3.0",
-    "@storybook/addon-interactions": "^7.3.0",
-    "@storybook/addon-mdx-gfm": "^7.3.0",
-    "@storybook/channels": "^7.3.0",
-    "@storybook/client-logger": "^7.3.0",
+    "@storybook/addon-actions": "0.0.0-pr-23778-sha-fa3b4d8f",
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply package.json:72]: I'm releasing an alpha version now:
https://github.com/storybookjs/storybook/pull/23926

**Pull request merged**